lib/rubocop/cop/base.rb

```diff
@@ -126,6 +127,36 @@ def self.support_multiple_source?
         false
       end
 
+      ## Gem requirements
+
+      # rubocop:disable Layout/ClassStructure
+      class << self
+        attr_reader :gem_requirements
+
+        # Register a version requirement for the given gem name.
+        # This cop will be skipped unless the target satisfies *all* requirements.
+        # @param [String] gem_name
+        # @param [String] version_requirement The version requirement,
+        #   using the same syntax as a Gemfile, e.g. ">= 1.2.3"
+        #
+        # @api public
+        def requires_gem(gem_name, version_requirement)
```

Comment:
Should this be in a mixin, like [`rubocop-rails`'s `TargetRailsVersion`](https://github.com/rubocop/rubocop-rails/blob/master/lib/rubocop/cop/mixin/target_rails_version.rb)?

It'll keep the logic nicely separated and tested in its own file, but it means [we couldn't use it from `target_rails_version_from_bundler_lock_file`](https://github.com/rubocop/rubocop/pull/12180) which is already in the Base.

Reply:
I'll have to think about this more, but I can live with the current proposal. 

There's also the question whether we want this to be able to work with multiple gems, but I'm guessing the likelihood that a cop depends on multiple gems is pretty tiny, so let's keep it simple for now.